scripts/install

```diff
@@ -1,202 +1,398 @@
-sed -i "s@LOGTOCHANGE@$final_logpath@g" ../conf/config.php
-sed -i "s@LOGTOCHANGE@$final_logpath@g" ../conf/logrotate
+#=================================================
+# CHECK IF THE APP CAN BE INSTALLED WITH THESE ARGS
+#=================================================
+
+### If the app uses nginx as web server (written in HTML/PHP in most cases), the final path should be "/var/www/$app".
+### If the app provides an internal web server (or uses another application server such as uwsgi), the final path should be "/opt/yunohost/$app"
+final_path=/usr/share/z-push
+config_path=/etc/z-push
+state_path=/var/lib/z-push
+log_path=/var/log/z-push
+#test ! -e "$final_path" || ynh_die "This path already contains a folder"
+
+# Normalize the url path syntax
+path_url=$(ynh_normalize_url_path $path_url)
+
+# Register (book) web path
+ynh_webpath_register $app $domain $path_url_activesync
+
+#=================================================
+# STORE SETTINGS FROM MANIFEST
+#=================================================
+
+ynh_app_setting_set $app domain $domain
+ynh_app_setting_set $app path $path_url
+#ynh_app_setting_set $app admin $admin
+ynh_app_setting_set $app is_public $is_public
+#ynh_app_setting_set $app language $language
+
+#=================================================
+# STANDARD MODIFICATIONS
+#=================================================
+# FIND AND OPEN A PORT
+#=================================================
+
+### Use these lines if you have to open a port for the application
+### `ynh_find_port` will find the first available port starting from the given port.
+### If you're not using these lines:
+###		- Remove the section "CLOSE A PORT" in the remove script
+
+# Find a free port
+#port=$(ynh_find_port 8095)
+# Open this port
+#yunohost firewall allow --no-upnp TCP $port 2>&1
+#ynh_app_setting_set $app port $port
+
+#=================================================
+# INSTALL DEPENDENCIES
+#=================================================
+
+### `ynh_install_app_dependencies` allows you to add any "apt" dependencies to the package.
+### Those deb packages will be installed as dependencies of this package.
+### If you're not using this helper:
+###		- Remove the section "REMOVE DEPENDENCIES" in the remove script
+###		- As well as the section "REINSTALL DEPENDENCIES" in the restore script
+###		- And the section "UPGRADE DEPENDENCIES" in the upgrade script
+
+echo "deb http://repo.z-hub.io/z-push:/final/Debian_9.0/ /" | tee /etc/apt/sources.list.d/z-push.list
+wget -qO - http://repo.z-hub.io/z-push:/final/Debian_9.0/Release.key | sudo apt-key add -
+yunohost tools update
+ynh_install_app_dependencies z-push-common z-push-backend-caldav z-push-backend-carddav z-push-backend-combined z-push-backend-imap z-push-backend-ldap z-push-backend-galsearch-ldap z-push-autodiscover
+ynh_install_app_dependencies php-memcached php-cli php-soap 
```

Comment:
Same here, try the helper

Reply:
I tryed without sucess:

[Z-push installation procedure](https://wiki.z-hub.io/display/ZP/Installation) ask to add repo
`deb http://repo.z-hub.io/z-push:/final/Debian_9.0/ /`

But when using `ynh_install_extra_repo --repo="deb http://repo.z-hub.io/z-push:/final/Debian_9.0/ /"`, there is an error: https://paste.yunohost.org/raw/facexifecu

Reply:
> absolute Suite Component

apt doesn't like the '/' for the suite.
If it was working before, have a look to the file when it was working, and the file with the helper. You should be able to fix it.

Reply:
With `ynh_install_extra_repo --repo="deb http://repo.z-hub.io/z-push:/final/Debian_9.0/ /" `

z-push content is `deb http://repo.z-hub.io/z-push:/final/Debian_9.0/ / http://repo.z-hub.io/z-push:/final/Debian_9.0/ /`

The only way I found to solve that was to directly use ynh_add_repo  and ynh_pin_repo, ugly but working

Reply:
As the helper is experimental, if you find a bug, it's better to fix it right now instead of use another way.

Reply:
Maybe the component part didn't work, https://github.com/YunoHost-Apps/Experimental_helpers/blob/master/ynh_add_secure_repos/ynh_add_secure_repos__2#L125
Probably because there's no space between.
By the way, you don't have any ?

Reply:
Ok, that's not really a deb repository. That's a strange repository of files.
Didn't even know Debian could work with that.

We can't really build a source.list file from it. So it's a good idea to use the 2 helpers as you did.

You should try the pin though, I'm not sure it will match your source list.

Reply:
Yes very weird repo... 
The pin seems working:
`ynh_pin_repo --package="*" --pin="origin \"repo.z-hub.io\"" --name="$app"`
give me a file /etc/apt/preferences.d/z-push with content:
`Package: *
Pin: origin repo.z-hub.io
Pin-Priority: 50
`

But i haven't checked for now if it the desired result

Reply:
I turned the pin to -1 and tried to install a package only available on this repo, z-push-config-apache. The pin worked as expected, the package wasn't available anymore.
So it's ok.

Reply:
Cool, thanks for your help